### PR TITLE
Bug 1838216 - mergify: require successful complete-push task for geckoview updates

### PR DESCRIPTION
**Pull request by jcristau**
Because the github-actions bot is untrusted its PR graphs don't include ui tests, so green complete-pr is not enough to avoid breakage.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1838216

### CODE REVIEW

**Review by rvandermeulen — APPROVED** [review]
Worth a shot! We can keep an eye on relbot PRs after this lands to make sure things still merge as expected.

File: .mergify.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ pull_request_rules:
     conditions:
       - author=github-actions[bot]
       - status-success=complete-pr
+      - status-success=complete-push
```
Comment (jonalmeida) [inline .mergify.yml:42]: Should this be an OR operator? By default, I think it's an AND (I might be wrong; I'll double-check again).

Comment (jcristau) [reply .mergify.yml:42]: The "and" is what I wanted here.

Comment (rvandermeulen) [reply .mergify.yml:42]: I'm OK with it being an AND, personally.

Comment (jonalmeida) [reply .mergify.yml:42]: Aye, LGTM then. 🙂

**Pull request merged**